### PR TITLE
feat(android)!: configurable SoundPool and `AudioManager.mode`

### LINKED ISSUE #135: Play sound with alarm stream
Android have multiple streams for sounds, media, alarm, etc.

Can you implement a new parameter for handle this ? I would like to play my sound like a alarm, to play the sound regardless of the current user's volume for media/songs.

Thanks in advance, and sorry for my bad english

### LINKED ISSUE #1240: earpieceOrSpeakersToggle is missing in the new version of the audioplayers
## Checklist
- [X] I made sure that the feature I am proposing doesn't already exist
- [X] I've tried the forceSpeaker and respectSilence configuration (AudioContextConfig) with no success

## Use case / Problem

In the version 0.20 there was an `earpieceOrSpeakersToggle` method. 
I use this method for an audio guide, when you can decide if listed the audio with earpiece or if there are not many people around you use the speaker.

## Proposal / Solution
Add the routing capabilities to the new `AudioContextConfig` if it's possible or reintroduce the `earpieceOrSpeakersToggle` method

### PR DESCRIPTION
**Pull request by Gustl22**
* feat(android): configurable SoundPool in LOW_LATENCY mode
* fix(android): apply global AudioContext to `MediaPlayer`
* feat(android): add `AudioMode` to AudioContextAndroid
* feat(android): set `AudioManager.mode` and `AudioManager.isSpeakerphoneOn` on `setGlobalAudioContext`
* docs: add missing docs for audio_context_config library
* docs: add docs on how to set platform-specific audio context

# Description

SoundPool could not be configured with AudioContext attributes. This Merge Request adds the functionality to also configure playing sources in LOW_LATENCY mode.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:`, `chore:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation and added dartdoc comments with `///`, where necessary.
- [x] I have updated/added relevant examples in [example].

## Breaking Change

- [x] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

### Migration instructions

If you create an AudioPlayer in `LOW_LATENCY` mode, it's presets are not the same as before as they are taken from the `AudioContext` now. Nether the less you can set them via:

```dart
final audioContext = const AudioContext(
  android: AudioContextAndroid(
    usageType: AndroidUsageType.game,
  ),
),
```

Set context globally, before creating a player:

```dart
AudioPlayer.global.setGlobalAudioContext(audioContext);
final player = AudioPlayer()..setPlayerMode(PlayerMode.lowLatency);
```

Set context for an existing single player:

```dart
player.setPlayerMode(PlayerMode.lowLatency)
player.setAudioContext(audioContext );
```

## Related Issues

Closes #1240
Closes #135

Related:
#1202
#1321
#934

[issue database]: https://github.com/bluefireteam/audioplayers/issues
[Contributor Guide]: https://github.com/bluefireteam/audioplayers/blob/main/contributing.md#feature-requests--prs
[Conventional Commit]: https://conventionalcommits.org
[example]: https://github.com/bluefireteam/audioplayers/tree/main/packages/audioplayers/example

### CODE REVIEW

File: packages/audioplayers_android/android/src/main/kotlin/xyz/luan/audioplayers/player/MediaPlayerPlayer.kt [hunk 1]
```diff
@@ -77,7 +79,7 @@ class MediaPlayerPlayer(
     }
 
     override fun updateContext(context: AudioContextAndroid) {
-        // TODO(luan) is this global?
+        // Set audioContext for single player
```
Comment (Gustl22) [inline packages/audioplayers_android/android/src/main/kotlin/xyz/luan/audioplayers/player/MediaPlayerPlayer.kt:82]: "for this player"

File: packages/audioplayers_android/android/src/main/kotlin/xyz/luan/audioplayers/player/SoundPoolPlayer.kt [hunk 2]
```diff
@@ -10,77 +9,42 @@ import xyz.luan.audioplayers.source.Source
 import xyz.luan.audioplayers.source.UrlSource
 import java.util.Collections.synchronizedMap
 
-// TODO(luan) make this configurable
-private const val MAX_STREAMS = 100
+// TODO(luan): make this configurable
+const val MAX_STREAMS = 100
```
Comment (Gustl22) [inline packages/audioplayers_android/android/src/main/kotlin/xyz/luan/audioplayers/player/SoundPoolPlayer.kt:13]: may can make private again

File: packages/audioplayers/example/lib/components/tgl.dart [hunk 3]
```diff
@@ -32,7 +32,7 @@ class Tgl extends StatelessWidget {
   }
 }
 
-class EnumTgl<T extends Enum> extends StatelessWidget {
+class EnumTgl<T extends Enum?> extends StatelessWidget {
```
Comment (Gustl22) [inline packages/audioplayers/example/lib/components/tgl.dart:35]: Need optional still?

File: packages/audioplayers/example/lib/tabs/audio_context.dart [hunk 4]
```diff
@@ -101,28 +121,126 @@ class _AudioContextTabState extends State<AudioContextTab>
...
+        ),
+        getIosOptionsCbx(AVAudioSessionOptions.mixWithOthers),
+        getIosOptionsCbx(
+          AVAudioSessionOptions.overrideMutedMicrophoneInterruption,
+        ),
       ],
     );
   }
 
+  Widget getIosOptionsCbx(AVAudioSessionOptions option) {
```
Comment (Gustl22) [inline packages/audioplayers/example/lib/tabs/audio_context.dart:227]: Make a widget

**Comment by Gustl22** [comment]
Consider adding the bullets in the PR description, when squashing.

**Review by spydon — APPROVED** [review]

**Pull request merged**